### PR TITLE
FIX : No usage of the function updateProduction in the update function

### PR DESCRIPTION
**Pull request by atm-adrien**
## FIX : No usage of the function updateProduction in the update function

This function deletes the lines contained in the MO concerned to then add them according to the lines of the associated BOM.
I can't really see the point?

Moreover, when I create an MO without any linked BOM, add lines and then modify data of the MO concerned, it deletes all the lines that I have previously created. This is due to the use of the updateProduction function in the MO update.

### CODE REVIEW

File: htdocs/mrp/class/mo.class.php [hunk 1]
```diff
@@ -612,11 +612,6 @@ public function update(User $user, $notrigger = false)
 			$error++;
 		}
 
-		$result = $this->updateProduction($user, $notrigger);
```
Comment (eldy) [inline htdocs/mrp/class/mo.class.php:615]: I think usage of this method here is done on purpose.
However, inside this method it seems to miss code under the comment 

// TODO Check that production has not started. If yes, we stop here.

There is nothing in this TODO. We call the method when we are in draft mode, but we should do nothing if production has started.

Comment (atm-adrien) [reply htdocs/mrp/class/mo.class.php:615]: I admit that I don't understand the need to delete the MO lines to add them later (without the changes made before the MO update).
Indeed, I understand the use of an update function for MO lines: It is possible to modify the quantity of an OF in "Draft" status, its lines must therefore be modified accordingly. However, this does not require deleting the rows and then adding them again.

Moreover, as I said above, when I create an MO without an associated BOM, add lines to it and then modify the MO (the label for example), all the lines previously entered are deleted. And it has nothing to do with whether production has started or not. This is due to the fact that we delete all the lines of the MO to add that of the associated BOM, but what happens to MOs that do not have associated BOMs?

I think the solution is to use a real MO rows update function (like it's done for all other objects in Dolibarr)

I'm developing a fix to correct that

**Comment by atm-adrien** [comment]
Travis say that there is an error, but the showed error does not correspond to my feature :
image

**Comment by atm-adrien** [comment]
@eldy

**Comment by atm-adrien** [comment]
Hi @eldy ! 

We really need to fix this bug for version 17, preferably before version 18 is released.

Do you have the possibility to merge this PR?

Please let me know if there is any problem.

**Pull request merged**